tools/tasks/task.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Base class for all tasks.
+ */
+export abstract class Task {
+  /**
+   * Override this task if you want to implement some custom
+   * task activation mechanism. By default each task will be always executed.
+   *
+   * @param {string[]} files A list of files changed since the previous watch.
+   */
+  precondition(files: string[]): boolean {
```

Comment:
Maybe it's too late for this one, but I'll follow NG2 pattern and use a `canRun` or something like that. Sorry, I was out today.